packages/lesswrong/components/posts/BookmarkButton.tsx

```diff
@@ -1,10 +1,19 @@
 import React from 'react';
 import { Components, registerComponent } from '../../lib/vulcan-lib/components';
-import { useBookmarkPost } from '../hooks/useBookmarkPost';
 import withErrorBoundary from '../common/withErrorBoundary';
 import classNames from 'classnames';
 import { isFriendlyUI } from '../../themes/forumTheme';
 import type { Placement as PopperPlacementType } from "popper.js"
+import { useBookmark } from '../hooks/useBookmark';
+
+export const BookmarkableCollectionNames = ["Posts", "Comments"] as const;
+
+export const isBookmarkableCollectionName = (
```

Comment:
This is what `TupleSet` is for.

```
const bookmarkableCollectionNames = new TupleSet(["Posts", "Comments"] as const);
```

Then `bookmarkableCollectionNames.has(name)` acts as a type guard automatically.

Reply:
(and you get the type for free as well.)